cogeo_mosaic/backends/s3.py

```diff
@@ -24,69 +28,71 @@ def __init__(
         mosaic_def: Optional[Union[MosaicJSON, Dict]] = None,
         client: Optional[boto3_session.client] = None,
     ):
+        """Initialize S3Backend."""
         self.client = client or boto3_session().client("s3")
         self.key = key
         self.bucket = bucket
 
         if mosaic_def is not None:
             self.mosaic_def = MosaicJSON(**dict(mosaic_def))
         else:
-            self.mosaic_def = self.read(bucket, key)
+            self.mosaic_def = self.read()
 
     def tile(self, x: int, y: int, z: int) -> Tuple[str]:
         """Retrieve assets for tile."""
-
         return get_assets_from_json(self.mosaic_def.tiles, self.quadkey_zoom, x, y, z)
 
     def point(self, lng: float, lat: float) -> Tuple[str]:
         """Retrieve assets for point."""
-
         tile = mercantile.tile(lng, lat, self.quadkey_zoom)
         return get_assets_from_json(
             self.mosaic_def.tiles, self.quadkey_zoom, tile.x, tile.y, tile.z
         )
 
     def write(self, gzip=None):
+        """Write mosaicjson document to AWS S3."""
         body = dict(self.mosaic_def)
         if gzip or (gzip is None and self.key.endswith(".gz")):
             body = _compress_gz_json(body)
+        else:
+            body = json.dumps(body).encode("utf-8")
 
-        _aws_put_data(
-            self.key, self.bucket, body, client=self.client,
-        )
```

Comment:
Again, I'm assuming black is picking up on a different line length that you have set